workshops/3D Heatmap/README.md

```diff
@@ -0,0 +1,273 @@
+import pandas as pd
+import seaborn as sns
+import numpy as np
+```
+Loading **csv** data in `data` variable
+
+```python
+data = pd.read_csv('volcano.csv')
+```
+
+Transform it to a long format to plot each data on the 3D plane. We will use  unstack and reset index functions.
+Store them in column x, y, z.
+
+```df``` denotes dataframe
+
+```python
+df = data.unstack().reset_index()
+df.columns = ['X', 'Y', 'Z']
+```
+
+And also transform the old column name into some numerical value using categorical function. Plots have numerical values on the axis of the plane.
+
+```python
+df['X'] = pd.Categorical(df['X'])
+df['X'] = df['X'].cat.codes
+```
+
+Make the plot in the 3D plane with the help of the projection function.
+Finally, trisurf function creates a surface by finding a set of triangles formed between adjacent points. 
+
+```python
+fig = plt.figure()
+ax = fig.gca(projection='3d')
+ax.plot_trisurf(df['Y'], df['X'], df['Z'], cmap=plt.cm.jet, linewidth=0.2)
+```
+`cmap=plt.cm.jet` returns the jet colormap as a three-column array. Each row in the array contains the red, green, and blue intensities for a specific color
+
+
+![Cmap](https://cloud-25g0giqzr.vercel.app/0colormap_jet.png)
+
+and then we will
+add a color bar that map values to colors. The point's height on the mountain increase the color value will shift from blue to red.
+
+```python
+surf = ax.plot_trisurf(df['Y'], df['X'], df['Z'], cmap=plt.cm.jet, linewidth=0.2)
+fig.colorbar(surf, shrink=0.5, aspect=5)
+```
+We will use this function to display the plot on our computer screen.
+```python
+plt.show()
+```
+---
+**Source Code**
+```python
+from mpl_toolkits.mplot3d import Axes3D
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+import numpy as np
+
+data = pd.read_csv('volcano.csv')
+
+df = data.unstack().reset_index()
+df.columns = ['X', 'Y', 'Z']
+
+df['X'] = pd.Categorical(df['X'])
+df['X'] = df['X'].cat.codes
+
+fig = plt.figure()
+ax = fig.gca(projection='3d')
+ax.plot_trisurf(df['Y'], df['X'], df['Z'], cmap=plt.cm.jet, linewidth=0.2)
+
+surf = ax.plot_trisurf(df['Y'], df['X'], df['Z'], cmap=plt.cm.jet, linewidth=0.2)
+fig.colorbar(surf, shrink=0.5, aspect=5)
+
+plt.show()
+```
+---
+**Output**
+![Volcano Heatmap](https://cloud-94iqxy8lo.vercel.app/0volcano.gif)
+
+To make 3D plots, we need each data point must have three coordinates to define it's position. These three are it's latitude, longitude, and altitude (X, Y, and Z). This data comes from the volcano.csv dataset. Then we will plot each point, and then it will give us the 3D model of the volcano. And then, we have also added colors to make this 3D plot into 3D Heatmap.
+
+Fantastic! You have created your first 3D Heatmap plot.
+
+# Voilà!
+
+Congratulations!✌️ you have completed the workshop. Share it with your friends and family.
+Share with everyone🤗✌️🥳👏🏅🌇🎊
+
+![congratulations g.i.f](https://cloud-1th3ydnib.vercel.app/2workshop_happy.gif)
+
+# Keep Hacking!
```

Comment:
Let's make these more human, give each one a paragraph. View Tunes for an example.